service/resource/configmap/desired.go

```diff
@@ -2,8 +2,58 @@ package configmap
 
 import (
 	"context"
+
+	"github.com/prometheus/prometheus/config"
+	"gopkg.in/yaml.v2"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/microerror"
+
+	"github.com/giantswarm/prometheus-config-controller/service/prometheus"
 )
 
 func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interface{}, error) {
-	return nil, nil
+	configMap, err := r.k8sClient.CoreV1().ConfigMaps(r.configMapNamespace).Get(
+		r.configMapName, metav1.GetOptions{},
+	)
```

Comment:
That smells. The config map is what should be mutated towards the desired state. When you compute the desired state you should not rely on the current state that you want to mutate. The only thing you need to compute the desired state are the targets that should be scraped and the secrets available for scrapable targets. I would expect to fetch the services and the secrets here to compute the desired state. What do I miss?

Reply:
The configuration file will/may include other scraping information, such as to scrape the host cluster kubelets, or global information about scrape timeouts.

We want to preserve this information. So, we add in the services that we want to scrape to the configuration file.

Reply:
I understand this. Just to have this said. It smells anyway, like the no-op magic. A cleaner design would be to have a config map completely managed by the controller resource and having a separate merger. Such merger could even be run by the controller but I dislike all these workarounds of the resource implementations.

Reply:
I can appreciate that.